plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -184,6 +232,9 @@ processSingleBinding = \case
     TermBind a s v@(VarDecl _ n _) rhs -> do
         maybeRhs' <- maybeAddSubst s n rhs
         pure $ TermBind a s v <$> maybeRhs'
+    TypeBind _ (TyVarDecl _ tn _) rhs -> do
+        modify' (extendType tn rhs)
```

Comment:
We need a Note about our approach to type bindings. In particular, we should justify why we're inlining them unconditionally. Shouldn't be too long, but good to write something and reference it from here.